go/store/nbs/journal.go

```diff
@@ -196,7 +195,7 @@ func trueUpBackingManifest(ctx context.Context, root hash.Hash, backing *journal
 	if err != nil {
 		return manifestContents{}, err
 	} else if !ok {
-		return manifestContents{}, fmt.Errorf("manifest not found when opening chunk journal")
+		return manifestContents{}, nil
```

Comment:
We changed the semantics here so that we can open a journal file without a manifest already existing.

We do this as part of `Clone` in some cases, where we copy the journal from a remote into our local store, and then we AddTableFilesToManifest to add it to our manifest. The new sanity checking logic wants to open the journal and read its chunks before we agree to add it to the store. This change allows us to do that.

The change below to `ParseIfExists` stitches it together, by continuing to go to the backing manifest as long as we have never loaded an actual manifest here.

Reply:
Probably worth a comment to that effect